### PR TITLE
Update parameter passing

### PR DESCRIPTION
**Pull request by rohithn1**
*Issue #, if available:*

*Description of changes:*

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [X] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [X] I certify that the changes I am introducing will be backward compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [X] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [X] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [X] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [X] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [X] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by rohithn1** [comment]
### Note: The following unit tests failed before my commits.
```
======================================================= short test summary info ========================================================
FAILED tests/unit/test_estimator.py::test_local_mode - ValueError: Failed to get the Region information from the default config. Plea...
FAILED tests/unit/test_estimator.py::test_local_mode_file_output_path - ValueError: Failed to get the Region information from the def...
FAILED tests/unit/test_estimator.py::test_insert_invalid_source_code_args - ValueError: Must setup local AWS configuration with a reg...
FAILED tests/unit/test_estimator.py::test_output_path_default_bucket_and_prefix_combinations - ValueError: Failed to get the Region i...
FAILED tests/unit/test_estimator.py::test_stage_user_code_in_s3_default_bucket_and_prefix_combinations[s3://output-bucket/output-prefix/output-prefix2-s3://code-bucket/code-prefix/code-prefix2-expected__without_user_input__with_default_bucket_and_default_prefix0-expected__without_user_input__with_default_bucket_only0-expected__with_user_input__with_default_bucket_and_prefix0-expected__with_user_input__with_default_bucket_only0]
FAILED tests/unit/test_estimator.py::test_stage_user_code_in_s3_default_bucket_and_prefix_combinations[s3://output-bucket/output-prefix/output-prefix2-None-expected__without_user_input__with_default_bucket_and_default_prefix1-expected__without_user_input__with_default_bucket_only1-expected__with_user_input__with_default_bucket_and_prefix1-expected__with_user_input__with_default_bucket_only1]
FAILED tests/unit/test_estimator.py::test_stage_user_code_in_s3_default_bucket_and_prefix_combinations[None-s3://code-bucket/code-prefix/code-prefix2-expected__without_user_input__with_default_bucket_and_default_prefix2-expected__without_user_input__with_default_bucket_only2-expected__with_user_input__with_default_bucket_and_prefix2-expected__with_user_input__with_default_bucket_only2]
FAILED tests/unit/test_estimator.py::test_stage_user_code_in_s3_default_bucket_and_prefix_combinations[None-None-expected__without_user_input__with_default_bucket_and_default_prefix3-expected__without_user_input__with_default_bucket_only3-expected__with_user_input__with_default_bucket_and_prefix3-expected__with_user_input__with_default_bucket_only3]
FAILED tests/unit/test_estimator.py::test_stage_user_code_in_s3_default_bucket_and_prefix_combinations[output_path4-s3://code-bucket/code-prefix/code-prefix2-expected__without_user_input__with_default_bucket_and_default_prefix4-expected__without_user_input__with_default_bucket_only4-expected__with_user_input__with_default_bucket_and_prefix4-expected__with_user_input__with_default_bucket_only4]
FAILED tests/unit/test_estimator.py::test_stage_user_code_in_s3_default_bucket_and_prefix_combinations[output_path5-None-expected__without_user_input__with_default_bucket_and_default_prefix5-expected__without_user_input__with_default_bucket_only5-expected__with_user_input__with_default_bucket_and_prefix5-expected__with_user_input__with_default_bucket_only5]
FAILED tests/unit/test_estimator.py::test_stage_user_code_in_s3_default_bucket_and_prefix_combinations[file://output-bucket/output-prefix/output-prefix2-s3://code-bucket/code-prefix/code-prefix2-expected__without_user_input__with_default_bucket_and_default_prefix6-expected__without_user_input__with_default_bucket_only6-expected__with_user_input__with_default_bucket_and_prefix6-expected__with_user_input__with_default_bucket_only6]
FAILED tests/unit/test_estimator.py::test_stage_user_code_in_s3_default_bucket_and_prefix_combinations[file://output-bucket/output-prefix/output-prefix2-None-expected__without_user_input__with_default_bucket_and_default_prefix7-expected__without_user_input__with_default_bucket_only7-expected__with_user_input__with_default_bucket_and_prefix7-expected__with_user_input__with_default_bucket_only7]
```

File: src/sagemaker/estimator.py [hunk 1]
```diff
@@ -3846,8 +3848,20 @@ def _distribution_configuration(self, distribution):
             # smdistributed strategy selected
             if get_mp_parameters(distribution):
                 distribution_config["mp_parameters"] = get_mp_parameters(distribution)
+            # first make sure torch_distributed is enabled if instance type is p5
+            torch_distributed_enabled = distribution.get("torch_distributed").get("enabled", False)
             smdistributed = distribution["smdistributed"]
             smdataparallel_enabled = smdistributed.get("dataparallel", {}).get("enabled", False)
+            p5_enabled = True if self.instance_type == "p5.48xlarge" else False
```
Comment (sheilaliuxl) [inline src/sagemaker/estimator.py:3855]: nit: `self.instance_type == "p5.48xlarge"` is already a `bool`
- If you really need to explicitly keep it as `bool`, use `bool(...)` i.e. the ctor.

File: src/sagemaker/estimator.py [hunk 2]
```diff
@@ -3846,8 +3848,20 @@ def _distribution_configuration(self, distribution):
...
+            img_uri = "" if self.image_uri == None else self.image_uri
```
Comment (sheilaliuxl) [inline src/sagemaker/estimator.py:3856]: nit: Using `is None` is preferred than `== None`
- Try `pylint $filename` and it'll show you the suggestions

File: src/sagemaker/estimator.py (hunk 1 above)
Comment (rohithn1) [reply src/sagemaker/estimator.py:3855]: I actually tried this initially but got issues. Not sure if it was specific to my machine but it was blocking for no reason which is why I wrote out the explicit if condition. Explicitly typecasting to bool should help actually, will address this.

File: src/sagemaker/estimator.py (hunk 2 above)
Comment (rohithn1) [reply src/sagemaker/estimator.py:3856]: Addressing this atm

**Pull request merged**